packages/grid/x-data-grid-premium/src/tests/clipboard.DataGridPremium.test.tsx

```diff
@@ -250,6 +250,34 @@ describe('<DataGridPremium /> - Clipboard', () => {
         expect(getCell(1, 2)).to.have.text(secondColumnValuesBeforePaste[1]);
         expect(getCell(2, 2)).to.have.text(secondColumnValuesBeforePaste[2]);
       });
+
+      it('should paste all values when they exceed `pageSize` and pagination disabled but fit into visible row area', async () => {
+        render(
+          <Test
+            rowLength={8}
+            colLength={4}
+            paginationModel={{ page: 1, pageSize: 5 }}
```

Comment:
To replicate the reported issue (the page size is 0-based)
```suggestion
            paginationModel={{ page: 0, pageSize: 5 }}
```